### ISSUE TITLE
Candidate matching broken after delphes@3.4.3pre11

### ISSUE THREAD

**Issue by tmadlener** [key4hep/k4SimDelphes#52]
The tests comparing the edm4hep and delphes outputs are broken with the current delphes master. Some bisecting points to [this commit](https://github.com/delphes/delphes/commit/0060caa43632829f62d1224ff9dd2bef500f7ed6) where things start to break (tested with root@6.22/06 and root@6.24/00). The tests break with something along the lines:
```console
[...] lots of the following

**** WARNING: No matching ReconstructedParticle was found for a Jet constituent
**** WARNING: No matching ReconstructedParticle was found for a Jet constituent
**** WARNING: No matching ReconstructedParticle was found for a Jet constituent
**** WARNING: No matching ReconstructedParticle was found for a Delphes muon
**** WARNING: No matching ReconstructedParticle was found for a Jet constituent
**** WARNING: No matching ReconstructedParticle was found for a Jet constituent
**** WARNING: No matching ReconstructedParticle was found for a Jet constituent

[...] and then further down

EVENT: 0
MC relations of candidate 0 are different between delphes and edm4hep output
+ seeds_file_cleanup
+ cat /home/tmadlener/work/k4SimDelphes/build_root_6.24.00/tests/random_seeds.2JdYYVG5
RANDOM SEED FOR DELPHES CARD = 5515
RANDOM SEED FOR PYTHIA CMD = 4435
```

So basically the comparison fails at the first element already.

Previously there were 0 warnings for `No matching ReconstructedParticle was found for a Jet constituent` and essentially 0 for Delphes muons (the ones that appear are probably related to #48). After the commit mentioned above, it seems that not a single candidate can be matched any longer. From the diff in the commit it is not immediately clear to me why the candidate matching breaks so badly.

Additionally, the matching seems to be completely broken with root@6.22/08, or at least I cannot even find a good commit to start bisecting on with that version (e.g. 3.4.3pre10 works with 6.22/06 and 6.24/00 but not with 6.22/08).

**Comment by tmadlener** [comment]
With the fixes in delphes also root@6.22/08 seems to be working fine again. Once this is fixed in delphes this can be closed.

**Comment by tmadlener** [comment]
Fixed in delphes.
